gridded_etl_tools/utils/metadata.py

```diff
@@ -793,6 +825,85 @@ def suppress_invalid_attributes(self, dataset: xr.Dataset):
             elif dataset.attrs[attr] is None:
                 dataset.attrs[attr] = ""
 
+    def modify_array_encoding(
```

Comment:
This function essentially does three different things: update, insert, and remove. Generally, I'm thinking it would best be three different functions, maybe with each function calling a common function (Python makes this a little awkward, since ideally the common function would also be a private function). I think in this case, since the tests are already well established, it's fine to leave as is.

Reply:
I've broken into two functions: update and remove. Insert and update use the exact same code so it feels excessive to treat them as if they're separate.